cpp/doc/DOCUMENTATION.md

```diff
@@ -0,0 +1,574 @@
+# libcuspatial C++ Documentation Guide
+
+These guidelines apply to documenting all libcuspatial C++ source files using doxygen style
+formatting although only public APIs and classes are actually 
+[published](https://docs.rapids.ai/api/libcuspatial/stable/index.html).
+
+## Copyright License
+
+The following is the license header comment that should appear at the beginning of every C++
+source file.
+
+```c++
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+```
+
+The comment should start with `/*` and not `/**` so it is not processed by doxygen.
+
+Also, here are the rules for the copyright year.
+
+- A new file should have the year in which it was created
+- A modified file should span the year it was created and the year it was modified (e.g.
+  `2019-2021`)
+
+Changing the copyright year may not be necessary if no content has changed (e.g. reformatting only).
+
+## Doxygen
+
+The [doxygen tool](http://www.doxygen.nl/manual/index.html) is used to generate HTML pages from the
+C++ comments in the source code. Doxygen recognizes and parses block comments and performs
+specialized output formatting when it encounters
+[doxygen commands](http://www.doxygen.nl/manual/commands.html).
+
+There are almost 200 commands (also called tags in this document) that doxygen recognizes in comment
+blocks. This document provides guidance on which commands/tags to use and how to use them in the
```

Comment:
```suggestion
blocks. This [descriptions](#descriptions) section provides guidance on which commands/tags to use and how to use them in the
```

Reply:
I don't see the section you are linking to. I think referring to "this document" is fine.